### ISSUE TITLE
[salt-ssh] Add a check for host in /etc/salt/roster

### ISSUE THREAD

**Issue by rallytime** [saltstack/salt#22332]
If you have a syntax error in your roster file for salt-ssh, such as a missing space after `host` such as `host:111.222.333.444` instead of `host: 111.222.333.444`, you get the following error:
```
# salt-ssh nt-cent6 test.ping
No hosts found with target nt-cent6 of type glob
```
YAML is probably treating the whole string as a key, so it's `host:111.222.333.444: None` or something like that.

There should be a check for host since it's a required member of the roster dictionary, and then provide a more helpful message if host isn't there, or you've mistyped something like this.

ping @basepi

**Comment by rallytime** [comment]
Oh, also, this is on the current HEAD of the 2014.7 branch:
```
# salt-ssh --versions-report
           Salt: 2014.7.4-50-g1be785e
         Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
         Jinja2: 2.7.3
       M2Crypto: 0.21.1
 msgpack-python: 0.4.4
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.11
          ioflo: Not Installed
          PyZMQ: 14.0.1
           RAET: Not Installed
            ZMQ: 4.0.4
           Mako: 0.9.1
```

**Comment by basepi** [comment]
Fixed in #23748. The render error was being logged in debug mode, but even there it was a terrible error. Bumped to error loglevel, and let the original yaml error through.
